deep_qa/data/instances/graph_align_instance.py

```diff
@@ -0,0 +1,262 @@
+from typing import Dict, List
+
+import numpy
+from overrides import overrides
+
+from .instance import TextInstance, IndexedInstance
+from ..data_indexer import DataIndexer
+from ...common.checks import ConfigurationError
+
+class GraphAlignInstance(TextInstance):
+    """
+        A ``GraphAlignInstance`` is a kind of ``TextInstance`` (for lack of knowing where else to put it)
+        that has a list of graphlets for each answer option, each of which contains one or more alignments.
+        Each alignment is a set of features (as floats).
+        These graphlets are designed to be used rapidly in pre-built TupleMatch systems, which can hopefully
+        learn which feature patterns are indicative of a correct answer (i.e., a good alignment to a knowledge
+        base sentence).
+
+        Parameters
+        ----------
+        answer_graphlets: List[ListList[List[float]]]]
+            This is a list whose length is equal to the num answer options, and each list element corresponds to an
+            answer option.  Each answer has a list of graphlets which come from features that model the alignment
+            between that question + answer option and a knowledge base sentence.
+
+        question_text: str, default=None
+            The original text of the question, if available.
+
+        label: int, default=None
+            The class label (i.e. the index of the correct multiple choice answer) -- corresponds to the
+            indices in self.answer_tuples.
+
+        index: int, default=None
+            The index of the question.
+        """
+    def __init__(self, answer_graphlets: List[List[List[List[float]]]], question_text: str=None,
+                 label: int=None, index: int=None):
+        super(GraphAlignInstance, self).__init__(label, index)
+        self.answer_graphlets = answer_graphlets
+        self.question_text = question_text
+
+    @overrides
+    def words(self) -> Dict[str, List[str]]:
+        # Unused
+        return None
```

Comment:
I think you need to return `{}` here instead of `None`, though if you have this actually running apparently you don't...